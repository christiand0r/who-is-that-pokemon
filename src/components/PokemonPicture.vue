<template>
  <div class="container">
    <div class="pokemon-image">
      <img
        :src="pokemonSource"
        alt="Pokémon"
        class="move-with-fade hidden-pokemon"
      />

      <img
        v-if="showPokemon"
        :src="pokemonSource"
        alt="Pokémon"
        class="fade-in"
      />
    </div>
  </div>
</template>

<script>
export default {
  name: "Pokemon Picture",
  props: {
    pokemonImage: {
      type: String,
      required: true,
      default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/7.svg",
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  computed: {
    pokemonSource() {
      return this.pokemonImage;
    },
  },
};
</script>

<style scoped>
img {
  left: 0;
  max-height: 320px;
  max-width: 320px;
  min-height: 320px;
  min-width: 320px;
  object-fit: contain;
  position: absolute;
  top: 0;
  user-select: none;

  -webkit-user-drag: none;
  z-index: 2;
}

.container {
  position: relative;
}

.container::after {
  background: url("@/assets/circle-spikes.png") no-repeat center center / 100%;
  bottom: -300px;
  content: "";
  height: 900px;
  left: -300px;
  position: absolute;
  width: 900px;
}

.hidden-pokemon {
  filter: brightness(0);
}

.pokemon-image {
  margin: 5rem 0 0 0;
  position: relative;
  min-width: 320px;
  min-height: 320px;
}
</style>
