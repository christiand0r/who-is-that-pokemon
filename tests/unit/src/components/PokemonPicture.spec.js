import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture.vue";

describe("PokemonPicture - Component", () => {
  test("debe ser igual que el snapshot", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        showPokemon: false,
        pokemonImage:
          "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/7.svg",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe mostrar la silueta del pokemon", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        showPokemon: false,
        pokemonImage:
          "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/7.svg",
      },
    });

    const [hiddenImage, image] = wrapper.findAll("img");

    //Image without filter should not load
    expect(image).toBeUndefined();

    //The silhouette must be shown
    expect(hiddenImage.exists()).toBeTruthy();

    //The silhouette source must be equals to pokemonImage
    expect(hiddenImage.attributes("src")).toBe(wrapper.vm.pokemonImage);

    //Check if the number before dot svg is the correct, don't need this test cause' pass the url complete
    //expect(hiddenImage.attributes("src")).toMatch("7.svg");

    //The silhouette must be have this class
    expect(hiddenImage.classes("hidden-pokemon")).toBeTruthy();
  });

  test("debe mostrar el pokemon si showPokemon: true", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        showPokemon: true,
        pokemonImage:
          "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/7.svg",
      },
    });

    const image = wrapper.find("img");

    //Image without filter must be shown
    expect(image.exists()).toBeTruthy();

    //The class hidden-pokemon shouldn't shown
    expect(image.classes("hidden-pokemon")).toBeFalsy();

    //The class fade-in must be shown
    expect(image.classes("fade-in")).toBeTruthy();
  });
});
