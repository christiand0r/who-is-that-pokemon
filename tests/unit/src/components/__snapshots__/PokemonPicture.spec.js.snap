// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture - Component debe ser igual que el snapshot 1`] = `
<div class="container">
  <div class="pokemon-image"><img alt="PokÃ©mon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/7.svg" class="move-with-fade hidden-pokemon"></div>
</div>
`;
